client/views/room/Header/Omnichannel/QuickActions/QuickActions.tsx

```diff
@@ -0,0 +1,208 @@
+	});
+
+	const requestTranscript = (email: string, subject: string): void => Meteor.call('livechat:requestTranscript', rid, email, subject, (err: any) => {
+		closeModal();
+		if (err != null) {
+			return handleError(err);
+		}
+		toastr.success(t('Livechat_transcript_has_been_requested'));
+		Session.set('openedRoom', null);
+	});
+
+	const sendTranscript = (email: string, subject: string, token: string): void => Meteor.call('livechat:sendTranscript', token, rid, email, subject, (err: any) => {
+		closeModal();
+		if (err != null) {
+			return handleError(err);
+		}
+
+		toastr.success(t('Your_email_has_been_queued_for_sending'));
+	});
+
+	const discardTranscript = (): void => Meteor.call('livechat:discardTranscript', rid, (error: any) => {
+		closeModal();
+		if (error != null) {
+			return handleError(error);
+		}
+		toastr.success(t('Livechat_transcript_request_has_been_canceled'));
+	});
+
+	const forwardChat = (departmentId?: string, userId?: string, comment?: string): void => {
+		if (departmentId && userId) {
+			return;
+		}
+		const transferData: { roomId: string; comment?: string; departmentId?: string; userId?: string } = {
+			roomId: rid,
+			comment,
+		};
+
+
+		if (departmentId) { transferData.departmentId = departmentId; }
+		if (userId) { transferData.userId = userId; }
+
+		Meteor.call('livechat:transfer', transferData, (error: any, result: any) => {
+			if (error) {
+				toastr.error(t(error.error));
+			} else if (result) {
+				closeModal();
+				toastr.success(t('Transferred'));
+				FlowRouter.go('/');
+			} else {
+				toastr.warning(t('No_available_agents_to_transfer'));
+			}
+		});
+	};
+
+	const confirmClose = (comment: string): void => Meteor.call('livechat:closeRoom', rid, comment, { clientAction: true }, (error: any) => {
+		closeModal();
+		if (error) {
+			return handleError(error);
+		}
+
+		toastr.success(t('Chat_closed_successfully'));
+	});
+
+	const openModal = useMutableCallback((id: string) => {
+		switch (id) {
+			case QuickActionsEnum.MoveQueue:
+				setModal(<ReturnChatQueueModal onMoveChat={moveChat} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.Transcript:
+				setModal(<TranscriptModal room={room} email={email} onRequest={requestTranscript} onSend={sendTranscript} onDiscard={discardTranscript} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.ChatForward:
+				setModal(<ForwardChatModal onForward={forwardChat} onCancel={closeModal} />);
+				break;
+			case QuickActionsEnum.CloseChat:
+				setModal(<CloseChatModal onConfirm={confirmClose} onCancel={closeModal} />);
+				break;
+			default:
+				break;
+		}
+	});
+
+	const actionDefault = useMutableCallback((e) => {
+		const index = e.currentTarget.getAttribute('data-quick-actions');
+		const { id } = actions[index];
+		openModal(id);
+	});
+
+	const hasManagerRole = useRole('livechat-manager');
+
+	const roomOpen = room && room.open && ((room.servedBy && room.servedBy._id === uid) || hasManagerRole);
```

Comment:
I realized `roomOpen` is always returning `false` and, from my tests, I noticed `room` object is not returning `servedBy` property, so the `if` statements will always be `false` if the user is not a `livechat-manager`.

Reply:
@rafaelblink an alternative here is to call `canAccessRoom` method.